### PR TITLE
[wip] add backup config controller

### PR DESCRIPTION
**Pull request by hexfusion**
ref: openshift/enhancements#415

### CODE REVIEW

**Comment by hexfusion** [comment]
... CI is not testing changes..

File: pkg/operator/etcd_assets/bindata.go [hunk 1]
```diff
@@ -66,6 +69,300 @@ func (fi bindataFileInfo) Sys() interface{} {
...
+              fieldPath: spec.nodeName
+      volumeMounts:
+        - mountPath: /etc/kubernetes/
+          name: etc-kubernetes
+      resources:
+        requests:
+          memory: 200M
+          cpu: 150m
+        limits:
+          memory: 200M
```
Comment (hexfusion) [inline pkg/operator/etcd_assets/bindata.go:179]: need to verify these

File: pkg/operator/starter.go [hunk 2]
```diff
@@ -14,20 +14,23 @@ import (
...
 	"github.com/openshift/library-go/pkg/operator/staticresourcecontroller"
 	"github.com/openshift/library-go/pkg/operator/status"
 	"github.com/openshift/library-go/pkg/operator/unsupportedconfigoverridescontroller"
 	"github.com/openshift/library-go/pkg/operator/v1helpers"
 
 	"github.com/openshift/cluster-etcd-operator/pkg/etcdcli"
 	"github.com/openshift/cluster-etcd-operator/pkg/etcdenvvar"
+	"github.com/openshift/cluster-etcd-operator/pkg/operator/backupconfigcontroller"
 	"github.com/openshift/cluster-etcd-operator/pkg/operator/bootstrapteardown"
+
```
Comment (hexfusion) [inline pkg/operator/starter.go:33]: drop newlines

File: pkg/resources/kubeapiserver/kubeapiserver.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
+package kubeapiserver
```
Comment (hexfusion) [inline pkg/resources/kubeapiserver/kubeapiserver.go:1]: this solution is temporary

File: pkg/operator/backupconfigcontroller/backupconfigcontroller.go [hunk 4]
```diff
@@ -0,0 +1,383 @@
...
+		} else {
+			args = append(args, fmt.Sprintf("%s- --cert-secrets=%s", YamlPad, s.Name))
+		}
+	}
+	return strings.Join(args, "\n")
+}
+
+func (c *BackupConfigController) manageBackupPod(substitutionReplacer *strings.Replacer, client coreclientv1.ConfigMapsGetter, recorder events.Recorder, operatorSpec *operatorv1.StaticPodOperatorSpec) (*corev1.ConfigMap, bool, error) {
+	podBytes := etcd_assets.MustAsset("etcd/backup-pod.yaml")
+	substitutedPodString := substitutionReplacer.Replace(string(podBytes))
```
Comment (ironcladlou) [inline pkg/operator/backupconfigcontroller/backupconfigcontroller.go:325]: Why not deserialize `backup-pod.yaml` into an actual `v1.Pod` object which we can manipulate in a sane way and then re-serialize?

Comment (hexfusion) [reply pkg/operator/backupconfigcontroller/backupconfigcontroller.go:325]: just time, to be honest, I will try to harden this up

**Pull request closed without merging**